gateway-service/src/main/java/org/zowe/apiml/gateway/filters/PageRedirectionFilterFactory.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.filters;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.EurekaClient;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.zowe.apiml.eurekaservice.client.util.EurekaMetadataParser;
+import org.zowe.apiml.product.gateway.GatewayClient;
+import org.zowe.apiml.product.routing.RoutedServices;
+import org.zowe.apiml.product.routing.ServiceType;
+import org.zowe.apiml.product.routing.transform.TransformService;
+import org.zowe.apiml.product.routing.transform.URLTransformationException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * PageRedirectionFilterFactory is a Spring Cloud Gateway Filter Factory that adapts a response from a routed service
+ * to handle 3xx status codes, and applies the headers to the response object.
+ *
+ */
+@Component
+@Slf4j
+public class PageRedirectionFilterFactory extends AbstractGatewayFilterFactory<PageRedirectionFilterFactory.Config> {
+
+    @Value("${server.attls.enabled:false}")
+    private boolean isAttlsEnabled;
+
+    private final EurekaClient eurekaClient;
+    private final EurekaMetadataParser metadataParser;
+    private TransformService transformService;
+
+    // TODO: solve multiple instances, there is not necessary to have multiple
```

Comment:
What about this TODO?

Reply:
I don't know the context of this TODO, @Shobhajayanna maybe?